src/Lister/SubPageList.php

```diff
@@ -91,13 +91,15 @@ private function renderForTitle( Title $title, array $parameters ) {
 	/**
 	 * @param Title $title
 	 * @param int $limit
+	 * @param string $sortOrder
 	 * @param bool $includeRedirects
 	 *
 	 * @return Page
 	 * @throws LogicException
 	 */
-	private function getPageHierarchy( Title $title, $limit, $includeRedirects ) {
+	private function getPageHierarchy( Title $title, $sortOrder, $limit, $includeRedirects ) {
 		$this->subPageFinder->setLimit( $limit );
+		$this->subPageFinder->setSortOrder( $sortOrder );
```

Comment:
It's cool you put in that check - it will make errors more apparent.

The issue here is that there is no abstraction between presentation layer, application logic and data access layer. If that is not clear to you, nvm, I'll take care of it then.